src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1223,6 +1227,41 @@ private function registerPropertyAccessConfiguration(array $config, ContainerBui
         ;
     }
 
+    private function registerAmqpConfiguration(array $config, ContainerBuilder $container, XmlFileLoader $loader)
+    {
+        $loader->load('amqp.xml');
+
+        $defaultConnectionName = $config['default_connection'] ?? null;
+
+        $match = false;
+        foreach ($config['connections'] as $name => $connection) {
+            $container
+                ->register("amqp.broker.$name", Broker::class)
+                ->setFactory(array(Broker::class, 'createWithDsn'))
+                ->addArgument($connection['dsn'])
+                ->addArgument($connection['queues'])
+                ->addArgument($connection['exchanges'])
+                ->setPublic(false)
+            ;
+
+            if (!$defaultConnectionName) {
+                $defaultConnectionName = $name;
+            }
+            if ($defaultConnectionName === $name) {
+                $match = true;
+            }
+        }
+
+        if (!$match) {
+            throw new \InvalidArgumentException(sprintf('The default_connection "%s" does not exist.', $defaultConnectionName));
+        }
+
+        $container
+            ->setAlias('amqp.broker', "amqp.broker.$defaultConnectionName")
+            ->setPublic(true)
```

Comment:
really?
how about an alias for autowiring also?

Reply:
It's quite common to publish a message from a controller or a command
About the alias: What should I alias ? the Broker::class to the **default** connection ?

Reply:
> What should I alias ? the Broker::class to the default connection ?

Yep 👍